### PR TITLE
Update API response

### PR DESCRIPTION
**Pull request by dhekra-rouatbi**
For more information on contributing to Tella source code, see the [Contributor Guidelines](contributing/contributor_guide.md).

## Update the API response

Add all the header fields to the API response

**Select the type of change(s) made in this pull request:**
- [ ] Bug fix *(Fixes an issue)*
- [X] New feature *(Adds functionality)*
- [ ] Documentation *(Fix to documentation)*

----------------------------------------------------------------------------------------

### CODE REVIEW

**Review by RobertShrestha — APPROVED** [review]
Everything looks good

File: Tella/Data/Networking/APICall/URLRequest.swift [hunk 1]
```diff
@@ -6,10 +6,13 @@
 import Foundation
 import Combine
 
+typealias APIResponse<Value> = AnyPublisher<(Value,[AnyHashable:Any]?), APIError>
```
Comment (ValbuenaG) [inline Tella/Data/Networking/APICall/URLRequest.swift:9]: I think we could make the names more self explanatory to distinguish between `APIResponse` and `APIDataResponse`. Maybe using `APIValueResponse` could enhance the readability of the code.

**Review by ValbuenaG — APPROVED** [review]
Great job Dhekra! I think this simplifies a lot the logic of the API response. I left only one comment but the overall it looks good to me

File: Tella/Data/Networking/APICall/URLRequest.swift (hunk 1 above)
Comment (dhekra-rouatbi) [reply Tella/Data/Networking/APICall/URLRequest.swift:9]: @ValbuenaG `Value` is a generic type, so we cannot rename the alias `APIValueResponse`

**Pull request merged**